### DISCUSSION TITLE
Support for body-fitted meshes

### DISCUSSION THREAD

**Discussion by gzagaris** [AMReX-Codes/amrex#4274] [General]
Hello,

Does AMReX provide support for body-fitted, curvilinear meshes?

We could probably store the X, Y, Z coordinates as a field on each patch, but, I think we would need to override the field-interpolation function accordingly. Is that something that is supported in AMReX?

Thank you for all your help.

**Comment by WeiqunZhang** [comment]
I think the ERF code (https://github.com/erf-model/ERF) has some support for terrain with variable dz's. @AMLattanzi can probably chime in.

**Comment by AMLattanzi** [comment]
The ERF code noted by Weiqun does employ terrain-following height coordinates. This coordinate system allows for mesh deformation only in the z-direction; see Eqs. 2-4 and Eqs. 8a-8f of [Sullivan (2014)](https://journals.ametsoc.org/view/journals/atsc/71/11/jas-d-14-0095.1.xml) for more details. In the ERF code, we store the physical z height and then compute all metric terms that enter the governing equations from that data structure  --- i.e., `J and \partial \xi_{i} / \partial x_{j}` in Eq. 8 of Sullivan.

If you wanted to do something more general, I believe you could store the `x, y, z` positions as you noted and then compute the associated metric terms. For this case, you would just have a dense Jacobian matrix (ours is rather sparse due to the parametrization in Eq. 2 of Sullivan).
